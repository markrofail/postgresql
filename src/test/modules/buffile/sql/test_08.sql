BEGIN;
SELECT buffile_create();

-- Neither disk space nor time needs to be wasted.
SET buffile_max_filesize TO 8192;

-- Write data across component file boundary and try to read it.
SELECT buffile_seek(0, 8190);
SELECT buffile_write('abcd');
SELECT buffile_seek(0, 8190);
SELECT buffile_read(8);
SELECT buffile_close();
COMMIT;
