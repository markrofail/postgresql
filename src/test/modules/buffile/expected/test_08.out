BEGIN;
SELECT buffile_create();
 buffile_create 
----------------
 
(1 row)

-- Neither disk space nor time needs to be wasted.
SET buffile_max_filesize TO 8192;
-- Write data across component file boundary and try to read it.
SELECT buffile_seek(0, 8190);
 buffile_seek 
--------------
            0
(1 row)

SELECT buffile_write('abcd');
 buffile_write 
---------------
             4
(1 row)

SELECT buffile_seek(0, 8190);
 buffile_seek 
--------------
            0
(1 row)

SELECT buffile_read(8);
 buffile_read 
--------------
 \x61626364
(1 row)

SELECT buffile_close();
 buffile_close 
---------------
 
(1 row)

COMMIT;
